openmc_models/inf_lat_dep/corewise/iter1/bcc_model.py

```diff
@@ -0,0 +1,114 @@
+import openmc
+import openmc.deplete
+import numpy as np
+
+
+#materials
+#graphite based on a3-3, triso layers pulled from reported values in
+#Neutronics characteristics of a 165 MWth Xe-100 reactor, Mulder et al
+
+#d_steps are (in days):
+#d_steps = [1] + [4] + [4] + [10]*9 + [25]*10 + [50]*24
+materials = openmc.Materials.from_xml('../materials.xml')
+dep_file = 'i0-dep-res.h5'
+res = openmc.deplete.Results(dep_file)
+dep_t = res.get_times()
+
+f = open('../nuc_list.txt', 'r')
+nuc_w_data = []
+for n in f.readlines():
+    nuc_w_data.append(n.strip())
+
+nuc_in_res = list(res[0].index_nuc.keys())
+
+nuclides = list(set(nuc_w_data) & set(nuc_in_res))
+
+print(nuclides)
+
+p_bins = [(0,19),(19,26),(26,31),(31,36),(36,41),(41,None)]
+p_comps = {'p01':{},
+           'p12':{},
+           'p23':{},
+           'p34':{},
+           'p45':{},
+           'p56':{}}
+p_t = [sum(dep_t[0:19]), 
+       sum(dep_t[19:26]), 
+       sum(dep_t[26:31]), 
+       sum(dep_t[31:36]),
+       sum(dep_t[36:41]), 
+       sum(dep_t[41:])]
+m_uco = materials[0].volume*materials[0].density
+
+for nuc in nuclides:
+    time, mass = res.get_mass('1', nuc, mass_units='g', time_units='d')
+    for i, k in enumerate(list(p_comps.keys())):
+        j = p_bins[i]
+        m = sum(mass[j[0]:j[1]]*(time[j[0]:j[1]]/p_t[i]))/m_uco
+        if m > 0:
+            p_comps[k][nuc] = m
+        else:
+            pass
+
+uco01= openmc.Material(name='UCO_01', material_id=10)
+uco01.set_density('g/cm3', 10.4)
+uco01.add_components(p_comps['p01'], percent_type = 'wo')
+
+uco12= openmc.Material(name='UCO_12', material_id=20)
+uco12.set_density('g/cm3', 10.4)
+uco12.add_components(p_comps['p12'], percent_type = 'wo')
+
+uco23= openmc.Material(name='UCO_23', material_id=30)
+uco23.set_density('g/cm3', 10.4)
+uco23.add_components(p_comps['p23'], percent_type = 'wo')
+
+uco34= openmc.Material(name='UCO_34', material_id=40)
+uco34.set_density('g/cm3', 10.4)
+uco34.add_components(p_comps['p34'], percent_type = 'wo')
+
+uco45= openmc.Material(name='UCO_45', material_id=50)
+uco45.set_density('g/cm3', 10.4)
+uco45.add_components(p_comps['p45'], percent_type = 'wo')
+
+uco56= openmc.Material(name='UCO_56', material_id=60)
+uco56.set_density('g/cm3', 10.4)
+uco56.add_components(p_comps['p56'], percent_type = 'wo')
```

Comment:
While there are more unnecessary reps here, it is nice to see p_comps is an indexed data structure. However, maybe just indexing by the number would be simpler (p_comps[1], p_comps[12])....